go/cmd/dolt/pprint/prettyprint.go

```diff
@@ -26,13 +26,80 @@ import (
 
 	"github.com/dolthub/go-mysql-server/sql"
 	"github.com/dolthub/vitess/go/sqltypes"
+	"github.com/fatih/color"
 
-	"github.com/dolthub/dolt/go/cmd/dolt/cli"
 	"github.com/dolthub/dolt/go/libraries/doltcore/table/untyped/csv"
 	"github.com/dolthub/dolt/go/libraries/doltcore/table/untyped/fwt"
 	"github.com/dolthub/dolt/go/libraries/utils/pipeline"
 )
 
+type ResultFormat byte
+
+const (
+	FormatTabular ResultFormat = iota
+	FormatCsv
+	FormatJson
+	FormatNull // used for profiling
+)
+
+func PrettyPrintResults(ctx *sql.Context, resultFormat ResultFormat, sqlSch sql.Schema, rowIter sql.RowIter) (rerr error) {
+	defer func() {
+		closeErr := rowIter.Close(ctx)
+		if rerr == nil && closeErr != nil {
+			rerr = closeErr
+		}
+	}()
+
+	if isOkResult(sqlSch) {
+		return printOKResult(rowIter)
+	}
+
+	// For some output formats, we want to convert everything to strings to be processed by the pipeline. For others,
+	// we want to leave types alone and let the writer figure out how to format it for output.
+	var p *pipeline.Pipeline
+	switch resultFormat {
+	case FormatCsv:
+		p = createCSVPipeline(ctx, sqlSch, rowIter)
+	case FormatJson:
+		p = createJSONPipeline(ctx, sqlSch, rowIter)
+	case FormatTabular:
+		p = createTabularPipeline(ctx, sqlSch, rowIter)
+	case FormatNull:
+		p = createNullPipeline(ctx, sqlSch, rowIter)
+	}
+
+	p.Start(ctx)
+	rerr = p.Wait()
+
+	return rerr
+}
+
+func printOKResult(iter sql.RowIter) (returnErr error) {
+	row, err := iter.Next()
+	if err != nil {
+		return err
+	}
+
+	if okResult, ok := row[0].(sql.OkResult); ok {
+		rowNoun := "row"
+		if okResult.RowsAffected != 1 {
+			rowNoun = "rows"
+		}
+
+		fmt.Fprintf(color.Output, "Query OK, %d %s affected\n", okResult.RowsAffected, rowNoun)
```

Comment:
Converted from cli.Print to Fmt.Fprint

Reply:
color.Output is not what you want here, or elsewhere. You need to use cli.Print methods to print. Otherwise this code is going to behave differently from other printing code when we change things in the CLI print methods.